backend/src/main/java/ai/verta/modeldb/versioning/VersioningServiceImpl.java

```diff
@@ -278,8 +286,230 @@ public void createCommit(
     }
   }
 
-  // TODO: add validation VR-3587
-  private void validateBlobDiffs(CreateCommitRequest request) throws ModelDBException {}
+  private void validateBlobDiffs(CreateCommitRequest request) throws ModelDBException {
```

Comment:
I'm not sure how to add different validation logic in generated code for each blob type with it. Maybe it will be easier if we'll ignore A blobs validation. And for diffs we'll just validate statuses. But I think there should be validation for A blobs too.

Reply:
We don't need to add validation in the generated code. The visitor is done outside and overrides what methods it wants. See https://github.com/VertaAI/modeldb/blob/feature/diffapi/backend/src/main/java/ai/verta/modeldb/versioning/blob/visitors/Validator.java for instance.

Reply:
moved validation to other branch, will follow up on that